source/views/sis/course-search/row.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import * as React from 'react'
+import {Text} from 'react-native'
+import type {CourseType} from '../../../lib/course-search/types'
+import {ListRow, Title} from '../../components/list'
+import {Row} from '../../components/layout'
+import {FormattedLine} from '../components/formatted-line'
+
+type Props = {
+	course: CourseType,
+	onPress: CourseType => any,
+}
+
+export class CourseRow extends React.PureComponent<Props> {
+	onPress = () => {
+		this.props.onPress(this.props.course)
+	}
+
+	render() {
+		const {course} = this.props
+		return (
+			<ListRow arrowPosition="center" onPress={this.onPress}>
+				<Row>
+					<Title lines={1}>{course.name}</Title>
+				</Row>
+				<Text>
+					{course.departments[0]} {course.number}
+					{course.section}
+				</Text>
+				{course.times && <FormattedLine items={course.times} />}
+				{course.instructors && <FormattedLine items={course.instructors} />}
+				{course.gereqs && <FormattedLine items={course.gereqs} />}
```

Comment:
For list height efficiency, please move this onto the same line as the deptnum.

```markdown
# Software Design
**CSCI 251A** (AQR, HWC)
Olaf Hall-Holt, Richard Brown
MWF 10:00am – 10:50am
Th 2:00pm – 5:00pm
```

`<Title>` for the title; bold for the deptnum; parenthesized GEs; comma-separated instructors; newline-delimited times, collapsed by day and formatted nicely with moment (and with en-dashes [–] between the times).

Reply:
Let’s leave GEs and times out of row results. I think the detail view should serve a greater purpose than just to show the locations and description. I would very much like to slim down what each row shows to 2 (max 3) lines

Reply:
I think that they're quite important to show in the list view. I don't want to have to tap and open a bunch of results to figure out which one works with my schedule.